lib/rspec/support/differ.rb

```diff
@@ -56,6 +58,26 @@ def diff_as_string(actual, expected)
       end
       # rubocop:enable Metrics/MethodLength
 
+      if defined?(RSpec::Mocks::ArgumentMatchers::AnyArgMatcher)
+        def diff_hashes_as_object(actual, expected)
+          expected_to_diff =
+            expected.reduce({}) do |hash, (key, value)|
+              if RSpec::Mocks::ArgumentMatchers::AnyArgMatcher === value
+                hash[key] = actual[key]
+              else
+                hash[key] = expected[key]
+              end
+              hash
+            end
+
+          diff_as_object(actual, expected_to_diff)
```

Comment:
If we inverted this, it would look like this:
```suggestion
          actual_to_diff =
            actual.reduce({}) do |hash, (key, value)|
              if RSpec::Mocks::ArgumentMatchers::AnyArgMatcher === expected(key)
                hash[key] = expected[key]
              else
                hash[key] = actual[key]
              end
              hash
            end

          diff_as_object(actual_to_diff, expected)
```